### ISSUE TITLE
Classic Gallery conversion to Block Gallery is not done fully.

### ISSUE THREAD

**Issue by arcmicro** [WordPress/gutenberg#30698]
## Description
When converting from Classic Gallery to Block Gallery the converted Gallery is missing the captions i wrote for images and image size selection, which was available in Classic post.

## Step-by-step reproduction instructions
1. Go to post in backend.
2. Click on Edit (Block Editor)
3. The post appear in editor.
4. then i select the classic content.
5. and then i press Convert to Blocks.
6. after this convert is done with missing image Captions i wrote and Size selection of images.

## Expected behavior
The expected result is that everything works as it was with Classic post editor.

## Actual behavior
The converted Gallery is missing the caption i wrote and image size selection.

## Screenshot
![Screenshot (2513)](https://user-images.githubusercontent.com/74606332/114266808-0685aa00-9a01-11eb-895e-0ce4b6e7a540.png)

## WordPress information
- WordPress version: 5.7
- Gutenberg version: it comes with last WP

- Are all plugins except Gutenberg deactivated? = "No"
- Are you using a default theme (e.g. Twenty Twenty-One)? = "No" GeneratePress last version.

## Device information
- Device: Desktop
- Operating system: Windows 10
- Browser: Epic based on Chromium 84.0

**Comment by paaljoachim** [comment]
Hei @iTesla13 

Thank you for creating the issue!

I am testing with WordPress 5.7.
Twenty Twenty One.
Gutenberg plugin 10.3.2.
Chrome

A local site.

Went to a post.
Added a Classic block.
Made a Gallery inside the Classic block.
Converted it to a block.
Captions are lost and the image size setting is also reset.


https://user-images.githubusercontent.com/5323259/114275180-d0a4ed80-9a21-11eb-881c-90b3b0decf58.mp4


-------------

Testing with the newly updated Gallery block.
https://make.wordpress.org/core/2021/03/23/gallery-block-update-call-for-testing/
One can download a special build of the Gutenberg plugin.

The error also happens in the new Gallery block build.
@glendaviesnz 


Associated wp.org support thread:
https://wordpress.org/support/topic/converting-classic-gallery-to-blocks-is-not-complete/#post-14302126

**Comment by arcmicro** [comment]
Hello @paaljoachim 
As you can see in my screenshot for me it even the Image Size Selection is not available after conversion to Gallery Block.
Maybe this is a theme issue but not sure, will test today with default or Astra theme to see if same and post back.
Thank you for your great help!

**Comment by paaljoachim** [comment]
I noticed that if I had selected for instance Medium for the Gallery images in Classic. On conversion to blocks the images would go to the default Large for blocks. So yes, I too noticed the change.

Btw I would assume the same result would be seen for really any theme.

**Comment by arcmicro** [comment]
I have test today with Default 2021 theme and both issues are present.

1. missing images caption i wrote
2. no image size selection settings

**Comment by glendaviesnz** [comment]
thanks @paaljoachim, I have added this to the gallery refactor project board

**Comment by glendaviesnz** [comment]
FYI, in terms of the refactored gallery block, because this is not a regression caused by that work, we are going to defer looking at this until after the first release of the refactored gallery code.

**Comment by arcmicro** [comment]
Hello, thank you for all the help.
I would like to report more bugs related to other types of Blocks in Gutenberg, but to open topics for each is too much to type and i must work, it would be easier to just comment them in some general Bug Report thread.

**Comment by paaljoachim** [comment]
Hi @iTesla13 Filip

What you can do is as you say add one report, and others like myself can create separate issues where it is needed.

**Comment by ssakcoR** [comment]
Hi,

I do experience the same bug. I want to convert more than 300 galleries from classic to Gutenberg, but due to the captions missing afterwards I hesitate to do that.

How long do you think this bug will take solving?

Would it be wise to convert the galleries now and expect the captions to appear after the update in the future? They are still visible for each individual image in Media Library. Is there a chance WP will retrieve them once this gets solved?

Thanks!

**Comment by arcmicro** [comment]
> Hi,
> 
> I do experience the same bug. I want to convert more than 300 galleries from classic to Gutenberg, but due to the captions missing afterwards I hesitate to do that.
> 
> How long do you think this bug will take solving?
> 
> Would it be wise to convert the galleries now and expect the captions to appear after the update in the future? They are still visible for each individual image in Media Library. Is there a chance WP will retrieve them once this gets solved?
> 
> Thanks!

No, better don't convert them, this looks irreversible, I still have some galleries to convert so it's better to wait.
Also the solving of bugs I think depends on how many people face this, so it will require more voices so the dev's fix it fast.

**Comment by youknowriad** [comment]
I think the issue is not fully solved, the size is now migrated properly but we still don't set the "caption" properly when creating image blocks by passing an "id". I think a potential solution is to use a "role": "local" attribute to indicate to the image block to fetch the "caption" from the image "id" when the image block is created from a gallery shortcode. The attribute name could be named "initFromId" or something like that and it could be a boolean.

**Comment by Mamaduka** [comment]
I think there's a much simpler fix for this problem. Here's the PR - #73748.
